### ISSUE TITLE
Scheduler crashes with `TypeError`

### ISSUE THREAD

**Issue by victorlin** [snakemake/snakemake#1389]
**Snakemake version**

Version 6.15.3 (latest available on bioconda)

**Describe the bug**

A specific workflow crashes with a `TypeError` stemming from the scheduler and its usage of `pulp.lpSum`.

- Running the workflow again picking up from where it left off, it will complete an additional step then fail with the same error.
- If I delete target files and start from scratch, it deterministically crashes at the same step.

**Logs**

```
[Tue Feb  8 20:19:42 2022]
Finished job 5.
16 of 32 steps (50%) done
Select jobs to execute...
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/conda-env-name/lib/python3.7/site-packages/snakemake/__init__.py", line 785, in snakemake
    keepmetadata=keep_metadata,
  File "/opt/homebrew/Caskroom/miniconda/base/envs/conda-env-name/lib/python3.7/site-packages/snakemake/workflow.py", line 1077, in execute
    success = self.scheduler.schedule()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/conda-env-name/lib/python3.7/site-packages/snakemake/scheduler.py", line 491, in schedule
    run = self.job_selector(needrun)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/conda-env-name/lib/python3.7/site-packages/snakemake/scheduler.py", line 740, in job_selector_ilp
    for job in jobs
  File "/opt/homebrew/Caskroom/miniconda/base/envs/conda-env-name/lib/python3.7/site-packages/pulp/pulp.py", line 2281, in lpSum
    return LpAffineExpression().addInPlace(vector)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/conda-env-name/lib/python3.7/site-packages/pulp/pulp.py", line 897, in addInPlace
    self.addInPlace(e)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/conda-env-name/lib/python3.7/site-packages/pulp/pulp.py", line 889, in addInPlace
    self.constant += other.constant
TypeError: unsupported operand type(s) for +=: 'int' and 'str'
```

Looks like something here is mixing `int` and `str`:

https://github.com/snakemake/snakemake/blob/41a5071b750dca5d7fceec324d81d9a93c86bdb6/snakemake/scheduler.py#L737-L742

**Minimal example**

Unfortunately, I am only able to reproduce this with a private workflow. However, see additional context below.

**Additional context**

The same workflow runs successfully with Snakemake versions 5.x.

I spot-checked several versions of Snakemake, and determined that the bug was introduced somewhere between 6.4.1 (runs fine) and 6.5.3 (crashes). I am unable to run this workflow on 6.5.0-6.5.2 due to #1069 which was fixed in 6.5.3. Could this be related?

Here's the diff between 6.4.1 and 6.5.3, in case it's handy: https://github.com/snakemake/snakemake/compare/v6.4.1...v6.5.3

**Comment by johanneskoester** [comment]
Good catch. A part of the fixed has been done in #1552. The other part comes in #1556.

**Comment by johanneskoester** [comment]
Thanks for reporting!
